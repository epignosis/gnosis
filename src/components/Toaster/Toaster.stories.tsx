import React from "react";
import { Story } from "@storybook/react";
import { toast } from "react-toastify";
import { Heading, Button } from "../../";
import Toaster from "./Toaster";
import { InfoSVG, DangerSVG, SuccessSVG, WarningSVG } from "@icons/core";

export default {
  component: Toaster,
  title: "Components/Toaster",
};

const ICONS = {
  info: <InfoSVG height={56} />,
  error: <DangerSVG height={56} />,
  success: <SuccessSVG height={56} />,
  warning: <WarningSVG height={56} />,
};

const publishToast = (type: string): void => {
  toast[type](
    <>
      {ICONS[type]}
      <div>
        <Heading size="md">Test</Heading>
        <span>Feature not available</span>
      </div>
    </>,
    { hideProgressBar: true },
  );
};

const Template: Story = () => (
  <>
    <Toaster />
    <div style={{ margin: "0 0 1rem" }}>
      <Button onClick={(): void => publishToast("info")}>Info toast</Button>
    </div>
    <div style={{ margin: "0 0 1rem" }}>
      <Button onClick={(): void => publishToast("success")}>Success toast</Button>
    </div>
    <div style={{ margin: "0 0 1rem" }}>
      <Button onClick={(): void => publishToast("warning")}>Warning toast</Button>
    </div>
    <div style={{ margin: "0 0 1rem" }}>
      <Button onClick={(): void => publishToast("error")}>Error toast</Button>
    </div>
  </>
);

export const Default = Template.bind({});

Default.parameters = {
  controls: { hideNoControlsWarning: true },
};
