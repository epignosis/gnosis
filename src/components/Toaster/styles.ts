import { css, Theme, SerializedStyles } from "@emotion/react";
import { hexToRGBA } from "@theme/default/colors";
import { mq } from "@theme/utils/breakpoints";

export const toaster = ({ alert }: Theme): SerializedStyles => {
  return css`
    padding: 1rem;

    ${mq["md"]} {
      padding: 0;
    }

    .close-btn {
      color: ${hexToRGBA(alert.closeBtnColor, 0.25)};
      margin: auto;
    }

    .Toastify__toast-body {
      display: flex;
      align-items: center;
      border-radius: 5px;

      svg {
        margin-right: 1rem;
      }
    }

    .Toastify__toast--info {
      background: ${alert.info.background};
      color: ${alert.info.color};
    }

    .Toastify__toast--success {
      background: ${alert.success.background};
      color: ${alert.success.color};
    }

    .Toastify__toast--warning {
      background: ${alert.warning.background};
      color: ${alert.warning.color};
    }

    .Toastify__toast--error {
      background: ${alert.danger.background};
      color: ${alert.danger.color};
    }
  `;
};
