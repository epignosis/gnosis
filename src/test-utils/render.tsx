import React, { FC, ReactElement, ComponentType } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options });

export * from "@testing-library/react";

export { customRender as render };
