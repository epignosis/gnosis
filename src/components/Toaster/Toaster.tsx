import React, { FC } from "react";
import { ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import { toaster } from "./styles";
import { CloseCircledSVG } from "@icons/core";

const Toaster: FC = () => {
  return (
    <ToastContainer
      position="top-right"
      css={toaster}
      closeButton={<CloseCircledSVG height={32} className="close-btn" />}
    />
  );
};

export default Toaster;
