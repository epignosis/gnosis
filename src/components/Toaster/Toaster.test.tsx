import React from "react";
import { toast } from "react-toastify";
import faker from "faker";
import userEvent from "@testing-library/user-event";
import Toaster from "./Toaster";
import { screen, render, waitFor } from "@test-utils/render";

describe("<Toaster/>", () => {
  it("renders correctly", () => {
    const toastTxt = faker.lorem.sentence();
    const btnTxt = faker.lorem.word();
    const publishToast = () => {
      toast(toastTxt);
    };

    render(
      <>
        <Toaster />
        <button onClick={publishToast}>{btnTxt}</button>
      </>,
    );

    userEvent.click(screen.getByRole("button"));

    waitFor(() => expect(screen.getByText(toastTxt)).toHaveTextContent(toastTxt));
  });

  it("matches snapshot", () => {
    const publishToast = () => {
      toast("test toast");
    };

    const { container } = render(
      <>
        <Toaster />
        <button onClick={publishToast}>test click</button>
      </>,
    );

    expect(container).toMatchSnapshot();
  });
});
