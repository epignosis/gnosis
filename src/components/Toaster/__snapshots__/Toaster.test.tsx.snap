// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toaster/> matches snapshot 1`] = `
<div>
  <div
    class="Toastify"
  />
  <button>
    test click
  </button>
</div>
`;
